README.md

```diff
@@ -36,80 +38,154 @@ bundle install
 WaterDrop is a complex tool, that contains multiple configuration options. To keep everything organized, all the configuration options were divided into two groups:
 
 - WaterDrop options - options directly related to Karafka framework and it's components
-- Ruby-Kafka driver options - options related to Ruby-Kafka/Delivery boy
+- Kafka driver options - options related to `Kafka`
+
+To apply all those configuration options, you need to create a producer instance and use the ```#setup``` method:
+
+```ruby
+producer = WaterDrop::Producer.new
+
+producer.setup do |config|
+  config.deliver = true
+  config.kafka = {
+    'bootstrap.servers': 'localhost:9092',
+    'request.required.acks': 1
+  }
+end
```

Comment:
this code will mutate `producer` object? WDYT if it returns a new object as a true FP lang? :D (Just idea, this API looks good for me)

Reply:
We can get a compromise - WDYT about freezing it after the configuration is done? (the config, not the producer as it needs to have buffers etc)

Reply:
good for me!